cannon/mipsevm/exec/mips_instructions.go

```diff
@@ -86,8 +86,10 @@ func ExecMipsCoreStepLogic(cpu *mipsevm.CpuScalars, registers *[32]uint32, memor
 			linkReg := uint32(0)
 			if fun == 9 {
 				linkReg = rdReg
+				stackTracker.PushStack(cpu.PC, rs)
+			} else {
+				stackTracker.PopStack()
 			}
-			stackTracker.PopStack()
```

Comment:
Wondering if we're handling the [other call to PushStack ](https://github.com/ethereum-optimism/optimism/blob/f93f9f40adcd448168c6ea27820aeee5da65fcbd/cannon/mipsevm/exec/mips_instructions.go#L25) correctly?  Looks like the jump instruction doesn't set a return address, so would we ever pop out of that operation?

Reply:
Both `j` and `jalr` instructions are typically coupled with a `jr` so we should eventually pop out of the operation. I say usually, because the Go runtime sometimes trampolines back via a `j` into its current stack without returning. But this doesn't happen often enough to bloat the depth of the stack we're tracking.

Reply:
I'd also imagine that the Go compiler would hardcode a destination when returning for some of these trampolines. But I haven't checked.